pkg/sql/sem/tree/format.go

```diff
@@ -365,7 +366,7 @@ func (ctx *FmtCtx) FormatNode(n NodeFormatter) {
 			ctx.WriteByte(')')
 		}
 	}
-	if f.HasFlags(fmtDisambiguateDatumTypes) {
+	if f.HasFlags(fmtDisambiguateDatumTypes) || f.HasFlags(FmtPGCatalog) {
```

Comment:
[nit] `if f.HasFlags(fmtDisambiguateDatumTypes | FmtPGCatalog) {`

Reply:
done

Reply:
Turns out `HasFlags` checks if _all_ of the flags are set, so I wrote a new `HasAnyFlags` function and used that.